x-pack/solutions/security/plugins/security_solution/public/sourcerer/containers/index.tsx

```diff
@@ -67,10 +67,11 @@ export const useSourcererDataView = (
     if (selectedDataView == null || missingPatterns.length > 0) {
       // old way of fetching indices, legacy timeline
       setLegacyPatterns(selectedPatterns);
-    } else {
+    } else if (legacyPatterns.length > 0) {
```

Comment:
sourcerer was unnecessarily re-rendering due to a reference change that took place because of an empty array being replaced with a new empty array. It could also be solved by pulling out a `const emptyArray = []` and re-using that but just went with the `if` check instead...